src/components/NewItemForm.jsx

```diff
@@ -55,14 +100,113 @@ const NewItemForm = () => {
   const [endpoint, setEndpoint] = useState("");
   const [password, setPassword] = useState("");
   const [username, setUserName] = useState("");
+  const [itemNameError, setItemNameError] = useState(clearError);
+  const [endpointError, setEndpointError] = useState(clearError);
+  const [passwordError, setPasswordError] = useState(clearError);
+  const [usernameError, setUserNameError] = useState(clearError);
+
   const [collectionID, setCollectionID] = useState("");
+  const [collectionArray, setCollectionArray] = useState([]);
+  const [defaultCollectionsID, setDefaultCollectionID] = useState("");
+
+  const [symmetricKey, setSymmetricKey] = useState("");
 
   const handleCollectionChange = (event) => {
     setCollectionID(event.target.value);
   };
 
-  const onSaveVaultItem = () => {
-    console.log("status");
+  // Gets vault collection and keys
+  const getUserDetails = async () => {
+    try {
+      let collectionResponse = await getCollections(getCollections);
+      let keyResponse = await getUserKey();
+      setCollectionArray(collectionResponse.data);
+
+      // We need to decrypt this in the future for vault items
+      let encryptedSymmetricKey = keyResponse.data.encrypted_symmetric_key;
+      for (let i = 0; i < collectionResponse.data.length; i++) {
+        let collection = collectionResponse.data[i];
+        if (collection.name == "Default") {
+          setDefaultCollectionID(collection.uuid);
+          setCollectionID(collection.uuid);
+        }
+      }
+    } catch (error) {
+      networkErrorHandler(error);
+    }
+  };
+
+  useEffect(() => {
+    getUserDetails();
+  }, []);
+
+  const onSaveVaultItem = async () => {
+    let errorStatus = false;
+    errorStatus = isNullCheck(itemName, setItemNameError);
+    errorStatus = isNullCheck(password, setPasswordError);
+    errorStatus = isNullCheck(endpoint, setEndpointError);
+    errorStatus = isNullCheck(username, setUserNameError);
+
+    if (errorStatus) {
+      return;
+    }
+
+    let vaultItemObject = {
+      name: itemName,
+      url: endpoint,
+      username: username,
+      password: password,
+    };
+
+    console.log(vaultItemObject);
+    // try {
+    //   let response = await postVaultItem(
+    //     vaultItemObject,
+    //     symmetricKey,
+    //     collectionID
+    //   );
+    // } catch (error) {
+    //   networkErrorHandler(error);
+    // }
```

Comment:
Is there a reason this is still commented out? Now that the symmetric key is in the global state (need to get latest from master), you should be able to access the key and then make a request to create vault items. I think this part is necessary for this PR to be complete.

Reply:
This should be updated now